.azure-pipelines/test_plan.py

```diff
@@ -139,24 +139,73 @@ def cancel(self, test_plan_id):
         print("Result of cancelling test plan at {}:".format(tp_url))
         print(str(resp["data"]))
 
-    def poll(self, test_plan_id, interval=60, timeout=36000):
-
-        print("Polling progress and status of test plan at https://www.testbed-tools.org/scheduler/testplan/{}" \
-              .format(test_plan_id))
-        print("Polling interval: {} seconds".format(interval))
-        print("Max polling time: {} seconds".format(timeout))
-
-        poll_url = "{}/test_plan/{}".format(self.url, test_plan_id)
+    def get_test_plan_status(self, test_plan_id):
+        url = "{}/test_plan/{}".format(self.url, test_plan_id)
         headers = {
             "Content-Type": "application/json"
         }
+
+        raw_resp = requests.get(url, headers=headers, timeout=10)
+        resp = raw_resp.json()
+
+        if not resp["success"]:
+            raise Exception("Query test plan at {} failed with error: {}".format(url, resp["errmsg"]))
+
+        resp_data = resp.get("data", None)
+        if not resp_data:
+            raise Exception("No valid data in response: {}".format(str(resp)))
+
+        status = resp_data.get("status", None)
+        result = resp_data.get("result", None)
+        resp_data = resp.get("data", None)
+
+        if not resp_data:
+            raise Exception("No valid data in response: {}".format(str(resp)))
+
+        return status, result, resp_data, url, raw_resp
+
+    def lock(self, test_plan_id, interval=60, timeout=36000):
```

Comment:
10 hours timeout?

Reply:
Another thing of this lock function is that in the function body, I only see code getting test plan status. I don't see any action to trigger the 'lock' action.